discord/guild/guild.py

```diff
@@ -0,0 +1,66 @@
+from typing import (
+    NamedTuple,
+    Optional,
+)
+
+from discord.abc.discordobject import DiscordObject
+from discord.channels.guildchannel import GuildChannel
+from discord.member.member import Member
+from discord.role.role import Role
+from discord.types.guildpayload import GuildPayload
+from discord.types.snowflake import Snowflake
+from discord.user.user import User
+
+
+class BanEntry(NamedTuple):
+    user: User
+    reason: Optional[str]
+
+
+class GuildLimit(NamedTuple):
+    filesize: int
+    emoji: int
+    channels: int
+    roles: int
+    categories: int
+    bitrate: int
+    stickers: int
+
+
+class Guild(DiscordObject):
+    __slots__ = (
+        'region'
+        'owner.id'
+        'mfa.level'
+        'name'
+        'id'
+        '_members'
+        '_channels'
+        '_vanity'
+        '_banner'
+    )
+
+    _roles: set[Role]
+    me: Member
+    owner_id: Snowflake
+
+    def __init__(self, data: GuildPayload):
+        self._members: dict[Snowflake, Member] = {}

```

Comment:
I assume these are for the cache and not directly for the guild? Are they misplaced?